lib/insights/api/common/application_controller_mixins/exception_handling.rb

```diff
@@ -35,6 +38,24 @@ def error_code_from_class(exception)
               DEFAULT_ERROR_CODE
             end
           end
+
+          def api_client_exception?(exc)
+            exc.respond_to?(:code) && exc.respond_to?(:response_body) && exc.respond_to?(:response_headers) &&
+              !exc.response_body.nil?
```

Comment:
`exc.respond_to?(:code) && exc.respond_to?(:response_body) && exc.respond_to?(:response_headers) &&
              !exc.response_body.nil?` Is there no other way than parsing all the `attr_reader`s to glean that this is the exception you're looking for?

Reply:
I was thinking about this as well - maybe we could have it just check the name of the class, for example `RBACApiClient::ApiError`/`ApprovalApiClient::ApiError` both would match `/Client::ApiError/`. Maybe we could have it do a string match on it? or would that not be close enough?

Reply:
@syncrou @lindgrenj6 
This exception class is a generic format for openapi_generated clients for examples for the 3 different clients we use 
Approval
https://github.com/RedHatInsights/approval-api-client-ruby/blob/5bdb4c7fecc44693ada76f93b8cd60d1a7c8f241/lib/approval-api-client-ruby/api_error.rb#L14
Topology
https://github.com/RedHatInsights/topological_inventory-api-client-ruby/blob/8c1448c7883b4f7f1c13660447a61025fa8ae5b0/lib/topological_inventory-api-client/api_error.rb#L15
RBAC
https://github.com/RedHatInsights/insights-rbac-api-client-ruby/blob/3d42dc6f6485627e6bbeb0955285b8f72a4a02d2/lib/rbac-api-client/api_error.rb#L15

Plus there is no guarantee that there would be a response body. When the API Client cannot establish a network connection we don't get back a response body. So checking for those 3 methods allows us to figure out that we have a valid api error object.